onnxruntime/core/session/abi_session_options_impl.h

```diff
@@ -14,6 +14,14 @@
 
 struct OrtSessionOptions {
   onnxruntime::SessionOptions value;
+
+  // This is only set when we provide an OrtSessionOptions to OrtEpFactory::CreateEp.
+  // In that scenario the InferenceSession has been created with the user provided OrtSessionOptions,
+  // so InferenceSession::GetSessionOptions() is the source of truth and existing_value is set to that.
+  // When set, `value` should be ignored.
+  // `const SessionOptions` as the expected usage in CreateEp does not involve modifying the value.
+  std::optional<const onnxruntime::SessionOptions*> existing_value;
```

Comment:
The compile API's `ModelCompilationOptions` makes a copy of the user's original session options. Should ModelCompilationOptions be modified to copy from `existing_value` if it is set?

Reply:
I don't think it needs to be aware. The only diff between the original OrtSessionOptions and InferenceSession::session_options_ would be the default ep_options from OrtEpDevice for selected EPs.